composition-js/src/__tests__/compose.demandControl.test.ts

```diff
@@ -0,0 +1,552 @@
+        enumWithCost: AorB
+        inputWithCost(someInput: InputTypeWithCost): Int
+      }
+    `,
+    { includeAllImports: true },
+  ),
+};
+
+const subgraphWithListSizeFromFederationSpec = {
+  name: 'subgraphWithListSize',
+  typeDefs: asFed2SubgraphDocument(
+    gql`
+      type Query {
+        fieldWithListSize: [String!] @listSize(assumedSize: 2000, requireOneSlicingArgument: false)
+      }
+    `,
+    { includeAllImports: true },
+  ),
+};
+
+const subgraphWithRenamedCostFromFederationSpec = {
+  name: 'subgraphWithCost',
+  typeDefs:
+    gql`
+      extend schema @link(url: "https://specs.apollo.dev/federation/v2.9", import: [{ name: "@cost", as: "@renamedCost" }])
+
+      enum AorB @renamedCost(weight: 15) {
+        A
+        B
+      }
+
+      input InputTypeWithCost {
+        somethingWithCost: Int @renamedCost(weight: 20)
+      }
+
+      type Query {
+        fieldWithCost: Int @renamedCost(weight: 5)
+        argWithCost(arg: Int @renamedCost(weight: 10)): Int
+        enumWithCost: AorB
+        inputWithCost(someInput: InputTypeWithCost): Int
+      }
+    `,
+};
+
+const subgraphWithRenamedListSizeFromFederationSpec = {
+  name: 'subgraphWithListSize',
+  typeDefs:
+    gql`
+      extend schema @link(url: "https://specs.apollo.dev/federation/v2.9", import: [{ name: "@listSize", as: "@renamedListSize" }])
+
+      type Query {
+        fieldWithListSize: [String!] @renamedListSize(assumedSize: 2000, requireOneSlicingArgument: false)
+      }
+    `,
+};
+
+// Used to test @cost applications on FIELD_DEFINITION
+function fieldWithCost(result: CompositionResult): FieldDefinition<ObjectType> | undefined {
+  return result
+    .schema
+    ?.schemaDefinition
+    .rootType('query')
+    ?.field('fieldWithCost');
+}
+
+// Used to test @cost applications on ARGUMENT_DEFINITION
+function argumentWithCost(result: CompositionResult): ArgumentDefinition<FieldDefinition<ObjectType>> | undefined {
+  return result
+    .schema
+    ?.schemaDefinition
+    .rootType('query')
+    ?.field('argWithCost')
+    ?.argument('arg');
+}
+
+// Used to test @cost applications on ENUM
+function enumWithCost(result: CompositionResult): EnumType | undefined {
+  return result
+    .schema
+    ?.schemaDefinition
+    .rootType('query')
+    ?.field('enumWithCost')
+    ?.type as EnumType;
+}
+
+// Used to test @cost applications on INPUT_FIELD_DEFINITION
+function inputWithCost(result: CompositionResult): InputObjectType | undefined {
+  return result
+    .schema
+    ?.schemaDefinition
+    .rootType('query')
+    ?.field('inputWithCost')
+    ?.argument('someInput')
+    ?.type as InputObjectType;
+}
+
+// Used to test @listSize applications on FIELD_DEFINITION
+function fieldWithListSize(result: CompositionResult): FieldDefinition<ObjectType> | undefined  {
+  return result
+    .schema
+    ?.schemaDefinition
+    .rootType('query')
+    ?.field('fieldWithListSize');
+}
+
+describe('demand control directive composition', () => {
```

Comment:
What happens when a directive is renamed in some, but not all of the subgraphs? I honestly can't remember if this works with other directives or not.

Reply:
Just pushed a test case for this. It results in a `LINK_IMPORT_NAME_MISMATCH` error